src/core-sdk-tasks/UpdateRuntimeConfig.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class UpdateRuntimeConfig : Task
+    {
+        [Required]
+        public ITaskItem[] RuntimeConfigPaths { get; set; }
+
+        [Required]
+        public string MicrosoftNetCoreAppVersion { get; set; }
+
+        [Required]
+        public string MicrosoftAspNetCoreAppVersion { get; set; }
+
+        public override bool Execute()
+        {
+            foreach (var file in RuntimeConfigPaths)
+            {
+                UpdateFile(file.ItemSpec);
+            }
+
+            return true;
+        }
+
+        private void UpdateFile(string file)
+        {
+            var text = File.ReadAllText(file);
+            JObject config = JObject.Parse(text);
+            var frameworks = config["runtimeOptions"]?["frameworks"];
```

Comment:
If there is only one framework, it will be in the `framework` element instead of having a list in the `frameworks`:

https://github.com/dotnet/sdk/blob/ffca47e9a36652da2e7041360f2201a2ba197194/src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs#L246-L253

Probably this task needs to handle both cases.